common/djangoapps/third_party_auth/tests/test_models.py

```diff
@@ -53,3 +53,144 @@ def test_clean_username_unicode_enabled(self):
         Test the username cleaner function with unicode enabled
         """
         assert clean_username('ItJüstWòrks™') == 'ItJüstWòrks'
+
+    def test_saml_configuration_dynamic_lookup(self):
+        """
+        Test that SAMLProviderConfig.get_config() properly retrieves the latest version of a SAML configuration
+        with matching slug rather than using the directly referenced version.
+
+        This tests how the edX SAMLConfiguration system handles updates:
+        - Updates to a configuration create a new record rather than modifying in place
+        - The system automatically retrieves the newest version with matching slug
+        - Even when all configs with matching slug are disabled, the system still uses the newest one
+        """
+        from django.utils import timezone
+        from datetime import timedelta
+        from ..models import SAMLConfiguration, SAMLProviderData, USE_LATEST_SAML_CONFIG
+        from openedx.core.djangoapps.site_configuration.tests.factories import SiteFactory
+        import mock
+
+        # ===== STEP 1: Setup initial configuration =====
```

Comment:
We don't need to add such detailed comments. Having verbose code is enough. If code gets ambiguous, then it's fine to add comment to highlight something.

Reply:
See https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules for more tips around commenting. Try to determine which comments are just restating what the code is already stating, and which are explaining intent or other necessary notes. This is an art, and not a science, so there is no perfect answer. I think we can possibly pair and work through this with others to discuss in more detail. Thank you.

I'm going to wait to review tests further until the other changes have come in.